std/algorithm/comparison.d

```diff
@@ -1007,16 +1011,28 @@ private:
     // Treat _matrix as a rectangular array
     ref CostType matrix(size_t row, size_t col) { return _matrix[row * cols + col]; }
 
-    void AllocMatrix(size_t r, size_t c) {
+    void AllocMatrix(size_t r, size_t c) @trusted {
         rows = r;
         cols = c;
         if (_matrix.length < r * c) {
-            delete _matrix;
-            _matrix = new CostType[r * c];
+            import core.stdc.stdlib : realloc;
+            auto m = cast(CostType *)realloc(_matrix.ptr, r * c * _matrix[0].sizeof);
+            if (!m)
+                assert(0);
+            _matrix = m[0 .. r * c];
             InitMatrix();
         }
     }
 
+    void FreeMatrix() @trusted {
+        import core.stdc.stdlib : free;
+
+        if (_matrix.ptr) {
```

Comment:
I'd say we're some 15 years past the last fix on those bugs. Walter please just skip the test, it just confuses future readers.